app/services/message_batch_import_service.rb

```diff
@@ -0,0 +1,37 @@
+require 'csv'
+
+class MessageBatchImportService
+  MessageTemplateNotFound = Class.new(StandardError)
+  MissingHeaders = Class.new(StandardError)
+  InvalidPhoneNumberException = Class.new(StandardError)
+
+  def import_message_batch(message_template_name, recipient_csv)
+    message_template = MessageTemplate.find_by(name: message_template_name)
+    raise MessageTemplateNotFound unless message_template
+
+    message_batch = MessageBatch.create(message_template: message_template)
+
+    created_count = 0
+    CSV.foreach(recipient_csv, headers: true, header_converters: :symbol) do |row|
+      raise MissingHeaders if [:case_id, :phone_number].difference(row.headers).any?
+      phone_number = convert_phone_number(row[:phone_number])
+      Recipient.create(program: 'SNAP', program_case_id: row[:case_id], phone_number: phone_number, message_batch: message_batch)
+      created_count += 1
+    end
+  end
+
+  private
+
+  def convert_phone_number(phone_number)
```

Comment:
This may need to be expanded later to also remove inner punctuation and whitespace